runner/lib/graph.py

```diff
@@ -40,6 +40,26 @@ def _missing_tasks(self):
         lst = [node._missing_dependencies() for node in self._nodes.values()]
         return set(itertools.chain.from_iterable(lst))
 
+    @staticmethod
+    def task_comparator(x, y):
+        """
+        A comparator which takes into account the common task name format:
+            <int:prefix>-<str:taskname>
+        """
+        if '-' in x and '-' in y:
+            x_pre = x.split('-')[0]
+            y_pre = y.split('-')[0]
+            if x_pre.isdigit() and y_pre.isdigit() and (x_pre != y_pre):
+                x = int(x_pre)
+                y = int(y_pre)
+                # double zeros are a special case: when compared to a single
+                # zero, it should be considered as "less." To make this work
+                # for any number of zeros, we'll convert them into negatives.
+                if (x == 0 and y == 0) and (len(x_pre) > 1 or len(y_pre) > 1):
```

Comment:
We still need a change IMO. I can take out the 00 part, but right now a task prefixed with 100, runs before a task prefixed with 99. Leaving in the split would fix that up.